### PR TITLE
feat: add history

### PR DESCRIPTION
**Pull request by antoncoding**
* Add history page
* Add new env variable L1 RPC
* History: Need to configure L2 optimismPortals for opstack chains

### CODE REVIEW

File: env.example [hunk 1]
```diff
@@ -1,3 +1,6 @@
 NEXT_PUBLIC_WALLETCONNECT_ID=xxxxxxxxxxxxxxxxxxxxxxxxxxx
 NEXT_PUBLIC_OPTIMISM_ETHERSCAN_API_KEY
-NEXT_PUBLIC_ARBITRUM_ETHERSCAN_API_KEY
\ No newline at end of file
+NEXT_PUBLIC_ARBITRUM_ETHERSCAN_API_KEY
+
+# l1 RPC
+NEXT_PUBLIC_L1_RPC_URL=https://eth-sepolia.g.alchemy.com/v2/EAcbL5Luk3laL5WwnguTm4OIiGmjX89X
```
Comment (NIC619) [inline env.example]: This is the public api key?

File: src/config/chainConfig.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+export const chainConfigs: Record<number, ChainConfig> = {
+  // [base.id]: {
+  //   portalAddress: '0x0000000000000000000000000000000000000000', // Replace with actual address
+  //   startBlock: 0, // Replace with actual start block
+  //   chain: base,
+  //   isOpstack: true
+  // },
+  [optimismSepolia.id]: {
+    portalAddress: '0x16Fc5058F25648194471939df75CF27A2fdC48BC', // Replace with actual address
+    startBlock: 4071248, // Replace with actual start block
```
Comment (NIC619) [inline src/config/chainConfig.ts]: Is it an L1 block? The start block of the deployment of the portal contract?

File: src/constants/abis.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+export const INBOX_ABI = [
```
Comment (NIC619) [inline src/constants/abis.ts]: Is this for Arbitrum or some other chains?

File: src/hooks/useForceInclusionHistory.ts [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+          const events = await l1Client.getLogs({
+            address: config.portalAddress,
+            event: DEPOSIT_EVENT,
+            args: {
+              from: address,
+            },
+            fromBlock: BigInt(config.startBlock),
+            toBlock: 'latest',
+          });
+
```
Comment (NIC619) [inline src/hooks/useForceInclusionHistory.ts]: The event query block range limit seems to be 5000 so we should probably just check the latest 5000 blocks (as `startBlock` tends to be way back in the past)?
```suggestion
          const EVENT_QUERY_BLOCK_RANGE = 5000 - 100; // Block range limit is 5000. We reduce it further by 100 as a buffer
          ...

          const latestBlockNumber = l1Client.getBlockNumber();
          const events = await l1Client.getLogs({
            address: config.portalAddress,
            event: DEPOSIT_EVENT,
            args: {
              from: address,
            },
            fromBlock: BigInt(latestBlockNumber - EVENT_QUERY_BLOCK_RANGE),
            toBlock: 'latest',
          });

```

File: env.example (hunk 1 above)
Comment (antoncoding) [reply env.example]: updated and revoked 🤕

File: src/config/chainConfig.ts (hunk 2 above)
Comment (antoncoding) [reply src/config/chainConfig.ts]: yes! added more comments

File: src/constants/abis.ts (hunk 3 above)
Comment (antoncoding) [reply src/constants/abis.ts]: Opps just old file forgot to remove

File: src/hooks/useForceInclusionHistory.ts (hunk 4 above)
Comment (antoncoding) [reply src/hooks/useForceInclusionHistory.ts]: updated, also added comment on the history page to show that we only fetch recent histories

File: src/pages/history/[address].tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+      <div className="min-h-[calc(100vh-64px)] w-full flex flex-col py-8">
+        <div className="flex-grow flex justify-center">
+          <div className="w-full max-w-2xl px-4">
+            <div className="bg-card rounded-lg shadow-lg p-6">
+              <div className="flex mb-6 flex-col gap-2">
+                <h1 className="text-2xl font-bold">History</h1>
+
+                <div className="flex items-center gap-2">
+                  <div className="text-sm text-gray-500">
+                    Force Inclusions in the last 5000 blocks:
```
Comment (NIC619) [inline src/pages/history/[address].tsx]: Just one last nit! 🤣🙏
```suggestion
                    Force Inclusions in the last 5000 L1 blocks:
```

**Review by NIC619 — APPROVED** [review]

**Pull request merged**